lib/ddtrace/workers.rb

```diff
@@ -71,6 +76,24 @@ def stop
         @run = false
       end
 
+      # Closes all available queues and waits for the trace and service buffer to flush
+      def shutdown!
+        return false if @shutting_down
+        @shutting_down = true
+        @trace_buffer.close
+        @service_buffer.close
+        sleep(0.1)
```

Comment:
do we need any sleep at all? if we close both buffers and we stop the worker, technically we should just `join` no? looking here: https://github.com/DataDog/dd-trace-rb/blob/47b0dd226f6828cc3d45905112d45bc1ce19986e/lib/ddtrace/workers.rb#L66-L70
both `callback_traces` (and `callback_services`):
* empties the buffer
* sends the buffers

So if we join we simply wait that the flushing thread exits the `while`, so for sure buffers are empty AND sent. The join has a timeout of 5 seconds, that is x5 times our flushing interval so it's reasonable high to ensure that traces are flushed.

Am I missing something?

Reply:
@ryplo since this approach works for Resque, we can merge the PR and make a new one to improve that code block.